### PR TITLE
Make --insecure and --host flags work when watching an execution.

### PR DESCRIPTION
**Pull request by jeevb**
# TL;DR

`-h/--host` flag does not get picked up for `watch-execution`:
```
> flyte-cli watch-execution -i -h localhost:51234 -u ex:sunflower-sandbox:default:f97312912bdfc476eab1
...
flytekit.common.exceptions.user.FlyteAssertion: No configuration set for [platform] url.  This is a required configuration.
```

`-i/--insecure` flag does not get picked up for `watch-execution`:
```
> FLYTE_PLATFORM_URL=localhost:51234 flyte-cli watch-execution -i -u ex:sunflower-sandbox:default:f37243487d6794178b76
...
E1003 02:24:42.122552000 4817231296 ssl_transport_security.cc:1439]    Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
...
```

However, this works:
```
> FLYTE_PLATFORM_URL=localhost:51234 FLYTE_PLATFORM_INSECURE=true flyte-cli watch-execution -u ex:sunflower-sandbox:default:f97312912bdfc476eab1
```

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Correctly patch the `platform_config.URL` and `platform_config.INSECURE` values from CLI arguments for `flyte-cli watch-execution`

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]
hmm, yeah i didn't think about this earlier sorry.  This should be done across the board for all flyte-cli subcommands but not right now.  Hopefully flytectl can replace flyte-cli entirely one day.  Thanks!
